bundles/framework/timeseries/instance.js

```diff
@@ -103,23 +103,52 @@ Oskari.clazz.define('Oskari.mapframework.bundle.timeseries.TimeseriesToolBundleI
             if (active) {
                 var conf = jQuery.extend(true, {}, this._controlPluginConf || {}, active.conf);
                 if (typeof conf.showControl === 'undefined' || conf.showControl) {
-                    this._createControlPlugin(active.delegate, conf);
+                    const controlClass = this._getControlPluginClazz(active.delegate);
+                    if (controlClass !== null) {
+                        this._createControlPlugin(controlClass, active.delegate, conf);
+                    }
                     return;
```

Comment:
If you move the return inside the if-statement that creates the plugin -> any possible existing ui will be removed if the new control wasn't created (`_removeControlPlugin()` is called at the end of the method). This should prevent some confusing side-effects where an existing UI might be used to control a timeseries layer that was configured with `ui="none"`
```suggestion
                    if (controlClass !== null) {
                        this._createControlPlugin(controlClass, active.delegate, conf);
                        return;
                    }
```